python/cudf/cudf/io/parquet.py

```diff
@@ -203,7 +202,11 @@ def read_parquet(
     for source in filepath_or_buffer:
         if ioutils.is_directory(source, **kwargs):
             fs = _ensure_filesystem(passed_filesystem=None, path=source)
-            source = stringify_path(source)
+            source = (
+                source.__fspath__()
+                if hasattr(source, "__fspath__")
+                else source
```

Comment:
I'll add this as a utility to `ioutils`. Is it acceptable to reuse the method name `stringify_path`.

Reply:
I think we should try to capture the behavior more in the name. Maybe `stringify_path_if_local`?

Reply:
I guess. The reason this was added was to address parsing `pathlib` objects (which is what pytest creates by default for `tmpdir` etc). Technically we don't want the string representation for local file objects either, just `path like` objects. How about `stringify_pathlike`?